app/backend/src/clj/teet/migration/user_permissions.clj

```diff
@@ -0,0 +1,71 @@
+(ns teet.migration.user-permissions
+  (:require [datomic.client.api :as d]
+            [teet.user.user-tx :as user-tx]))
+
+
+;; Pull users with global permissions that have no valid until
+;; {:admin 1 :manager 2 :internal-consultant 3 :external-consultant 4} ---
+;; sort by role importance and take first
+;; create transaction with call to (set-global-role) for all users with their hightest permission role
+
+(def permission-importance
+  {:admin 1
+   :manager 2
+   :internal-consultant 3
+   :external-consultant 4})
+
+(defn remove-overlapping-global-rights
```

Comment:
Here as well, very hard to parse the implementation, and not sure what's meant by overlapping global rights. So docstring and maybe a few helpful comments in the threading macro.